lib/nviz/exag.c

```diff
@@ -63,6 +64,8 @@ int Nviz_get_exag_height(double *val, double *min, double *max)
 
     G_debug(1, "Nviz_get_exag_height(): value = %f min = %f max = %f", *val,
             min ? *min : 0.0, max ? *max : 0.0);
+    if (nsurfs > 0)
+        G_free(surf_list);
```

Comment:
This is what I meant:

```suggestion
    G_free(surf_list);
```

Reply:
sorry i got confused with the word "should".
changes done in both the functions

Reply:
No worries, maybe I wasn't clear.
(GS_get_surf_list returns malloced array only if `nsurfs > 0`, otherwise NULL, so the check was really about that. Remember it is safe to free() a NULL value.)